### PR TITLE
Common Components Validations + Active Permit MHR Correction Scenario

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:*

*Description of changes:*
 /bcgov/entity#19515
- Display HomeLocation information with Active Permit Details on Location step (When active Permit)
- Associated validation and Review Confirm updates

 /bcgov/entity#19536
- Common Component Validation fixes for Civic Address and Home Location Types for both Transport Permits and MHR's
- DocId validation rule update as per ticket to include REQUIRED always (ie not just when the flow validation prompts)
 
 /bcgov/entity#19871
- Minor refactors to Business Lookup to allow tooltip for SPs
- Minor styling adjustments to meet design requirements

 
Screenshot 2024-03-18 at 10 28 36 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue [hunk 1]
```diff
@@ -430,9 +440,6 @@ export default defineComponent({
       localState.legalDescription = ''
       localState.locationInfo = resetLocationInfoFields(localState.locationInfo)
     })
-    watch(() => localState.validate, () => {
-      validateForms()
-    })
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeLocationType.vue:435]: Was a duplicate, obsolete after TP refactors (changed to prop)

File: ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue [hunk 2]
```diff
@@ -211,8 +211,9 @@ export default defineComponent({
       // Clear fields when country changes
       addressLocal.value.street = ''
       addressLocal.value.city = ''
-      addressLocal.value.region = ''
+      addressLocal.value.region = null
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeLocation/HomeCivicAddress.vue:214]: For v-select values, defaults of `null` are required to put the select in an unselected state, empty strings are considered a value.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1784-y8ai9prf.web.app

**Review by dimak1 — APPROVED** [review]
Looks good!

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Comment by cameron-eyds** [comment]
Added the Tooltip and Disabled fix for SPs in the Business Lookup, if anyone would like to re-review. Will leave the pr open a bit 👍

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1784-y8ai9prf.web.app

**Comment by dimak1** [comment]
> Added the Tooltip and Disabled fix for SPs in the Business Lookup, if anyone would like to re-review. Will leave the pr open a bit 👍

Still looks good 👍🏻

**Pull request merged**